inference-engine/tests/functional/inference_engine/transformations/compare_functions_test.cpp

```diff
@@ -208,4 +208,127 @@ TEST(TransformationTests, CompareFunctoinsTINegative) {
+        << "Return: >" << res.second << "< instead";
+}
+
+TEST(TransformationTests, ConstantNegativeDifferentShapes) {
+    const auto& f1 = [] {
+        using namespace ngraph::opset5;
+        auto constant = Constant::create(ngraph::element::f32, ngraph::Shape{1}, {1.0});
+
+        return std::make_shared<ngraph::Function>(
+            ngraph::NodeVector{constant}, ngraph::ParameterVector{});
+    }();
+
+    const auto& f2 = [] {
+        using namespace ngraph::opset5;
+        auto constant = Constant::create(ngraph::element::f32, ngraph::Shape{2}, {1.0});
+
+        return std::make_shared<ngraph::Function>(
+            ngraph::NodeVector{constant}, ngraph::ParameterVector{});
+    }();
+
+    const auto& res = compare_functions(f1, f2, false, false, false, true, true);
+    EXPECT_FALSE(res.first);
+    EXPECT_TRUE(contains(res.second, "mismatch in value: 'shape'"))
+        << "Return: >" << res.second << "< instead";
+}
+
+TEST(TransformationTests, ClampNegativeDifferentMin) {
+    const auto& f1 = [] {
+        using namespace ngraph::opset5;
+        auto constant = Constant::create(ngraph::element::f32, ngraph::Shape{1}, {1.0});
+        auto clamp = std::make_shared<Clamp>(constant, 1., 2.);
+
+        return std::make_shared<ngraph::Function>(
+            ngraph::NodeVector{clamp}, ngraph::ParameterVector{});
+    }();
+
+    const auto& f2 = [] {
+        using namespace ngraph::opset5;
+        auto constant = Constant::create(ngraph::element::f32, ngraph::Shape{1}, {1.0});
+        auto clamp = std::make_shared<Clamp>(constant, 1.3, 2.);
+
+        return std::make_shared<ngraph::Function>(
+            ngraph::NodeVector{clamp}, ngraph::ParameterVector{});
+    }();
+
+    const auto& res = compare_functions(f1, f2, false, false, false, true, true);
+    EXPECT_FALSE(res.first);
+    EXPECT_TRUE(contains(res.second, "mismatch in value: 'min'"))
+        << "Return: >" << res.second << "< instead";
+}
+
+TEST(TransformationTests, ClampNegativeDifferentMax) {
+    const auto& f1 = [] {
+        using namespace ngraph::opset5;
+        auto constant = Constant::create(ngraph::element::f32, ngraph::Shape{1}, {1.0});
+        auto clamp = std::make_shared<Clamp>(constant, 1., 2.);
+
+        return std::make_shared<ngraph::Function>(
+            ngraph::NodeVector{clamp}, ngraph::ParameterVector{});
+    }();
+
+    const auto& f2 = [] {
+        using namespace ngraph::opset5;
+        auto constant = Constant::create(ngraph::element::f32, ngraph::Shape{1}, {1.0});
+        auto clamp = std::make_shared<Clamp>(constant, 1., 200.);
+
+        return std::make_shared<ngraph::Function>(
+            ngraph::NodeVector{clamp}, ngraph::ParameterVector{});
+    }();
+
+    const auto& res = compare_functions(f1, f2, false, false, false, true, true);
+    EXPECT_FALSE(res.first);
+    EXPECT_TRUE(contains(res.second, "mismatch in value: 'max'"))
+        << "Return: >" << res.second << "< instead";
+}
```

Comment:
Can we add more tests for other attribute types?

Reply:
👍